platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/internal/ValidationProblemSerialization.java

```diff
@@ -78,6 +87,7 @@ public static GsonBuilder createGsonBuilder() {
         gsonBuilder.registerTypeHierarchyAdapter(DocLink.class, new DocLinkAdapter());
         gsonBuilder.registerTypeHierarchyAdapter(ProblemLocation.class, new LocationAdapter());
         gsonBuilder.registerTypeHierarchyAdapter(ProblemCategory.class, new ProblemCategoryAdapter());
+        gsonBuilder.registerTypeHierarchyAdapter(AdditionalData.class, new AdditionalDataAdapter());
```

Comment:
🤔 I didn't see this code before. Why are we using JSON for ValidationProblemSerialization while using Kryo for most other things?

Reply:
Thank you for the review!
I am not sure. The JSON-based serialisation was already there when we got started with the Problems API integration.

Reply:
It might be interesting to find out what is going on there.

Reply:
I introduced the JSON part, it used to be just a file that we would read on the other end.
with no structured information.

Reply:
https://github.com/gradle/gradle/issues/29188